src/components/TextField/TextField.tsx

```diff
@@ -20,23 +20,23 @@ export const TextField: React.FC<Props> = ({
   label = name,
   required = false,
   onChange = () => {},
-  validation,
+  validate = (inputValue) => inputValue,
 }) => {
-  // generage a unique id once on component load
   const [id] = useState(() => `${name}-${getRandomDigits()}`);
-
-  // To show errors only if the field was touched (onBlur)
   const [touched, setToched] = useState(false);
-  const [valid, setValid] = useState<boolean | void>(true);
-  const hasError = touched && required && (!value || !valid);
+  const [valid, setValid] = useState<boolean | string>(true);
+  const hasError = touched
+    && required
+    && (
+      (!value || value.split('').every(char => char === ' '))
+      || !valid
+    );
```

Comment:
This code looks bad. 
```suggestion
    && (
      !value.trim() || !valid
    );
```